core/src/main/java/org/apache/druid/data/input/impl/TsvInputFormat.java

```diff
@@ -27,15 +27,34 @@
 
 public class TsvInputFormat extends SeparateValueInputFormat
 {
+
+  private static Format getFormat(String delimiter)
+  {
+    if (delimiter != null && delimiter.length() > 0) {
+      Format.CustomizeSV.setDelimiter(delimiter, null);
+      return Format.CustomizeSV;
+    } else {
+      return Format.TSV;
+    }
+  }
+
   @JsonCreator
   public TsvInputFormat(
       @JsonProperty("columns") @Nullable List<String> columns,
       @JsonProperty("listDelimiter") @Nullable String listDelimiter,
+      @JsonProperty("delimiter") @Nullable String delimiter,
```

Comment:
Modifying `TsvInputFormat` to support a custom delimiter is awkward, since "TSV" implies that the delimiter is a tab. How about using an approach similar to `DelimitedParser` and `AbstractFlatTextFormatParser` instead? For example, the current implementation here is introducing regressions not present in `DelimitedParser` (e.g., `delimeter` and `listDelimeter` should be checked to not have the same value).

Reply:
Good point. I would suggest to rename the type of this input format. How about [`dsv`](https://en.wikipedia.org/wiki/Delimiter-separated_values) or `delimited`?

Reply:
https://druid.apache.org/docs/latest/ingestion/data-formats.html#tsv-delimited-parsespec, However in this documentation, `format: tsv` implies user could use any delimiter, and default is tab.
How about this design: rename `SeparateValueInputFormat` to `DelimitedInputFormat`,  when user say `tsv`, instantiate a `DelimitedInputFormat` object, when user say `csv`, instantiate `CSVInputFormat` which extends `DelimitedInputFormat`

Reply:
I think its original name was not good. We don't necessarily stick to the existing name, but can change it if there's a better one. 

> How about this design: rename `SeparateValueInputFormat` to `DelimitedInputFormat`, when user say `tsv`, instantiate a `DelimitedInputFormat` object, when user say `csv`, instantiate `CSVInputFormat` which extends `DelimitedInputFormat`

Do you mean you want to make `DelimitedInputFormat` as a concrete class? What is difference between `DelimitedInputFormat` and `CSVInputFormat` in that case?

Reply:
`DelimitedInputFormat ` would be used to support any delimiter (default is tab), ie:  when user say `tsv` as format.
`CSVInputFormat ` would be used to support strictly comma as delimiter,  ie: when user say `csv` as format

Reply:
Thanks, it sounds good to me. But I think its type should be something else instead of `tsv`.